lightning/src/chain/package.rs

```diff
@@ -381,57 +381,53 @@ impl PackageSolvingData {
 	fn finalize_input<Signer: Sign>(&self, bumped_tx: &mut Transaction, i: usize, onchain_handler: &mut OnchainTxHandler<Signer>) -> bool {
 		match self {
 			PackageSolvingData::RevokedOutput(ref outp) => {
-				if let Ok(chan_keys) = TxCreationKeys::derive_new(&onchain_handler.secp_ctx, &outp.per_commitment_point, &outp.counterparty_delayed_payment_base_key, &outp.counterparty_htlc_base_key, &onchain_handler.signer.pubkeys().revocation_basepoint, &onchain_handler.signer.pubkeys().htlc_basepoint) {
-					let witness_script = chan_utils::get_revokeable_redeemscript(&chan_keys.revocation_key, outp.on_counterparty_tx_csv, &chan_keys.broadcaster_delayed_payment_key);
-					//TODO: should we panic on signer failure ?
```

Comment:
Maybe these TODOs can be addressed while being here? AFAICT, we'd currently panic anyways two step up:
https://github.com/lightningdevkit/rust-lightning/blob/52edb351575ec997c414a66df0142f46a39aeba3/lightning/src/chain/onchaintx.rs#L484

So maybe the `bool` return value here and even the `Option` in the return of `finalize_malleable_package` could be removed?